### PR TITLE
Fix defaults in Configurations.md

### PR DESCRIPTION
**Pull request by theotherphil**
Fixes https://github.com/rust-lang-nursery/rustfmt/issues/1679 and https://github.com/rust-lang-nursery/rustfmt/issues/1722

### CODE REVIEW

File: Configurations.md [hunk 1]
```diff
@@ -1839,7 +1839,7 @@ See also: [`where_density`](#where_density), [`where_layout`](#where_layout), [`
 
 Overall strategy for where clauses
 
-- **Default value**: `"Default"`
+- **Default value**: `"Rfc"`
 - **Possible values**: `"Default"`, `"Rfc"`
```
Comment (nrc) [inline Configurations.md]: `Default` has changed to `Legacy` for several of these options.

**Comment by theotherphil** [comment]
I've updated the names for `where_style`. The only places the Style enum is used in this file are `where_style` and `control_style`. The latter isn't documented at all. I'll create a separate issue for fixing that.

(The only other options specified in config.rs but not mentioned in configurations.md are `verbose`, `file_lines`, `report_todo` and `report_fixme`, none of which affect formatting.)

**Comment by theotherphil** [comment]
I've updated this pull request to include adding `control_style` to Configurations.md

**Comment by theotherphil** [comment]
Hmm, the behaviour I've documented as expected appears to be described as a problem here: https://github.com/rust-lang-nursery/rustfmt/issues/1690

**Comment by nrc** [comment]
So for `control_style` I believe the difference is that with legacy we would visual indent the condition and with Rfc we block indent. With Rfc the opening brace goes on a newline unless the the last line of the conditional is just a closing paren/brace/bracket.

E.g.,

```
    if real_total <= limit && !pre_line_comments &&
        !items.into_iter().any(|item| item.as_ref().is_multiline())
    {
        DefinitiveListTactic::Horizontal
    }
```

**Comment by theotherphil** [comment]
Ok, thanks. I'll update the examples after work today.

**Comment by theotherphil** [comment]
I've now fixed the control_style example.

**Comment by nrc** [comment]
Great, thank you!

**Pull request merged**